pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,944 @@
+    POWER_ON = 128
+
+    #: This bit is set when an incorrect HP-IB code is sent to the power meter. For example,
+    # the command “QX” is a command error.
+    COMMAND_ERROR = 32
+
+    #: This bit is set when incorrect data is sent to the power meter. For example, the command
+    # “FR-3GZ” is an execution error.
+    EXECUTION_ERROR = 16
+
+    #: This bit is set true whenever a measurement error (error 1-49) occurs.
+    DEVICE_DEPENDENT_ERROR = 8
+
+
+Errors = {
+    1: "Power meter cannot zero the sensor",
+    5: "Power meter cannot calibrate sensor",
+    11: "Input overload on sensor",
+    15: "Sensor’s zero reference has drifted negative",
+    17: "Input power on sensor is too high for current range",
+    21: "Power reading over high limit",
+    23: "Power reading under low limit",
+    31: "No sensor connected to the input",
+    33: "Both front and rear sensor inputs, have sensors connected (Option 002 or Option 003 only)",
+    50: "Entered cal factor is out of range",
+    51: "Entered offset is out of range",
+    52: "Entered range number is out of range",
+    54: "Entered recall register number is out of range",
+    55: "Entered storage register number is out of range",
+    56: "Entered reference cal factor is out of range",
+    57: "RAM ID check failure",
+    61: "Stack RAM failure",
+    62: "ROM checksum failure",
+    64: "RAM failure",
+    65: "Analog I/O PIA Failure",
+    66: "Keyboard and Display PIA Failure",
+    67: "Analog-to-Digital converter Failure",
+    68: "HP-IB failure",
+    69: "Timer failure",
+    70: "Keyboard/Display controller failure",
+    71: "Keyboard data failure",
+    72: "Data line to A3U26 is open.",
+    73: "Keyboard/Display controller self-test failure",
+    74: "Display not responding",
+    75: "Digital failure",
+}
+
+
+class StatusMessage:
+    MeasurementErrorCode = (0, 1)
+    EntryErrorCode = (2, 2)
+    OperatingMode = (4, 2)
+    AutomaticRangeStatus = (6, 1)
+    Range = (7, 1)
+    # 8, 9 unused
+    AutoFilterStatus = (10, 1)
+    Filter = (11, 1)
+    # 12, 13 unused
+    LinearLogStatus = (14, 1)
+    # A
+    PowerRefStatus = (16, 1)
+    RelativeModeStatus = (17, 1)
+    TriggerMode = (18, 1)
+    GroupTriggerMode = (19, 1)
+    LimitsCheckingStatus = (20, 1)
+    LimitsStatus = (21, 1)
+    # 22 unused
+    OffsetStatus = (23, 1)
+    DutyCycleStatus = (24, 1)
+    MeasurementUnits = (25, 1)
+
+
+class HP437B(Instrument):
+    """Represents the HP437B Power Meters.
+
+    .. note::
+        Most command descriptions are taken from the document:
+        'Operating Manual 437B Power Meter'
+    """
+
+    def __init__(self, adapter, name="Hewlett-Packard HP437B", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            send_end=True,
+            **kwargs,
+        )
+
+    def _getstatus(status_type, modifier=lambda v: v):
```

Comment:
> THought about that already, that wouldn't work:
  get_process=_getstatus(StatusMessage.DutyCycleStatus), E   TypeError: 'staticmethod' object is not callable

The following does not work?
```suggestion
    @staticmethod
    def _getstatus(status_type, modifier=lambda v: v):
```

I made a small minimum working example and it works for me:
~~~python
from pymeasure.instruments import Instrument
from pymeasure.adapters.protocol import ProtocolAdapter

class Whatever(Instrument):

    @staticmethod
    def static(f):
        def xy(v):
            return f * v
        return xy

    prop = Instrument.control("G", "S", "docs", get_process=static(2))


inst = Whatever(ProtocolAdapter([("G", "1")]), name="abc")
p = inst.prop
assert p == 2
~~~

Reply:
I will fiddle around later, but the issue occured during pytest execution.

Reply:
An alternative is to move that method outside of the class.